### PR TITLE
Emit non-internal errors when `setBlockSize` is used outside of a GPU loop

### PR DESCRIPTION
**Pull request by DanilaFe**
A user coming from CUDA might think that they need to set the block size using a function call before calling the loop:

```Chapel
  setBlockSize(1024);
  foreach a in A do a+= 1;
```

However, `setBlockSize` is actually a special function that the compiler recognizes and handles. In the case above, the function is not in a format that the compiler expects, resulting in an internal error. This PR simply improves the error messages to explain what's wrong. 

Reviewed by @e-kayrakli -- thanks!

## Testing
- [x] `test/gpu/native`

### CODE REVIEW

File: compiler/optimizations/gpuTransforms.cpp [hunk 1]
```diff
@@ -1615,6 +1616,34 @@ static void cleanupTaskIndependentCapturePrimitives() {
...
+      USR_FATAL_CONT(callExpr, "'setBlockSize' must only be used in bodies of GPU-eligible loops");
+      explainAnchor = callExpr;
+
+      debuggerBreakHere();
+
+      // Search forward to try find the CForLoop corresponding to the GPU loop.
+      auto search = callExpr->next;
+      while (search) {
+        auto condStmt = toCondStmt(search);
+        if (gpuBranches.count(condStmt) > 0) {
```
Comment (e-kayrakli) [inline compiler/optimizations/gpuTransforms.cpp:1632]: Just so I am following the logic here: we are finding the next conditional from the offending call with the assumption that it is the dynamic check we introduce per kernel launch site?

File: test/gpu/native/errors/badblocksize.good [hunk 2]
```diff
@@ -0,0 +1,2 @@
+badblocksize.chpl:6: error: 'setBlockSize' must only be used in bodies of GPU-eligible loops
```
Comment (e-kayrakli) [inline test/gpu/native/errors/badblocksize.good:1]: Definitely not the authority here, but should this be "can only be used"?

**Review by e-kayrakli — APPROVED** [review]

File: compiler/optimizations/gpuTransforms.cpp (hunk 1 above)
Comment (DanilaFe) [reply compiler/optimizations/gpuTransforms.cpp:1632]: Yep! This is a little heuristic: if we match the pattern, we can print a nicer error message in cases like the original motiviating case:

```Chapel
setBlockSize(128);
foreach 1..100 { ... }
```

Comment (e-kayrakli) [reply compiler/optimizations/gpuTransforms.cpp:1630]: Understood. Could you update the comment to refer to the fact that we are actually looking for a conditional in the implementation and what that conditional is?

**Pull request merged**